source/FreeRTOS_UDP_IPv6.c

```diff
@@ -130,66 +130,26 @@ static eARPLookupResult_t prvStartLookup( NetworkBufferDescriptor_t * const pxNe
                                           BaseType_t * pxLostBuffer )
 {
     eARPLookupResult_t eReturned = eARPCacheMiss;
-    uint32_t ulIPAddress;
 
-    /* MISRA Ref 11.3.1 [Misaligned access] */
-    /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
-    /* coverity[misra_c_2012_rule_11_3_violation] */
-    const UDPPacket_t * pxUDPPacket = ( ( const UDPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
+    FreeRTOS_printf( ( "Looking up %pip with%s end-point\n",
+                       ( void * ) pxNetworkBuffer->xIPAddress.xIP_IPv6.ucBytes,
+                       ( pxNetworkBuffer->pxEndPoint != NULL ) ? "" : "out" ) );
 
-    if( pxUDPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE )
+    if( pxNetworkBuffer->pxEndPoint == NULL )
     {
-        FreeRTOS_printf( ( "Looking up %pip with%s end-point\n",
-                           ( void * ) pxNetworkBuffer->xIPAddress.xIP_IPv6.ucBytes,
-                           ( pxNetworkBuffer->pxEndPoint != NULL ) ? "" : "out" ) );
-
-        if( pxNetworkBuffer->pxEndPoint == NULL )
-        {
-            IPv6_Type_t eTargetType = xIPv6_GetIPType( &( pxNetworkBuffer->xIPAddress.xIP_IPv6 ) );
-            BaseType_t xIsGlobal = ( eTargetType == eIPv6_Global ) ? pdTRUE : pdFALSE;
-            pxNetworkBuffer->pxEndPoint = pxGetEndpoint( ( BaseType_t ) ipTYPE_IPv6, xIsGlobal );
-            FreeRTOS_printf( ( "prvStartLookup: Got an end-point: %s\n", pxNetworkBuffer->pxEndPoint ? "yes" : "no" ) );
-        }
-
-        if( pxNetworkBuffer->pxEndPoint != NULL )
-        {
-            vNDSendNeighbourSolicitation( pxNetworkBuffer, &( pxNetworkBuffer->xIPAddress.xIP_IPv6 ) );
-
-            /* pxNetworkBuffer has been sent and released.
-             * Make sure it won't be used again.. */
-            *pxLostBuffer = pdTRUE;
-        }
+        IPv6_Type_t eTargetType = xIPv6_GetIPType( &( pxNetworkBuffer->xIPAddress.xIP_IPv6 ) );
+        BaseType_t xIsGlobal = ( eTargetType == eIPv6_Global ) ? pdTRUE : pdFALSE;
+        pxNetworkBuffer->pxEndPoint = pxGetEndpoint( ( BaseType_t ) ipTYPE_IPv6, xIsGlobal );
+        FreeRTOS_printf( ( "prvStartLookup: Got an end-point: %s\n", pxNetworkBuffer->pxEndPoint ? "yes" : "no" ) );
     }
-    else
-    {
-        ulIPAddress = pxNetworkBuffer->xIPAddress.ulIP_IPv4;
```

Comment:
Why are we removing IPv4 handling

Reply:
It's duplicated dead code that is never executed. [This check](https://github.com/tony-josi-aws/FreeRTOS-Plus-TCP/blob/a497d48f1f5af632999bd52777d00707ec9a3b1c/source/FreeRTOS_UDP_IP.c#L108-L109) makes sure IPv4 handling is never required there.

Reply:
Was this not caught in UT coverage test?

Reply:
As the initial check was in a different file, no.